main.tf

```diff
@@ -0,0 +1,143 @@
+#####
+# Backup Vault
+#####
+resource "aws_backup_vault" "main" {
+  count = var.vault_name != null ? 1 : 0
```

Comment:
I have a feeling that this module will fail if you create backup vault outside the module and then pass it into the module `vault_name = aws_backup_vault.external_vault.name` . Can you please test that behaviour?

Reply:
Tested with the following:

Test 1:
1. Created a backup vault manually in AWS.
2. Ran an example without `vault_name` (i.e no backup vault is being created) and instead `target_vault_name` was set to the manually created backup vault. 

**Success**

Test 2:
1. Created a backup vault manually in AWS.
2. Created a data source for the vault in AWS.
3. ```target_vault_name =  data.aws_backup_vault.main.name"```
4. Again, ensured no backup vault was created.

**Success**

Reply:
What happens when you create all resources through terraform where vault is created outside the module and you reference it as follows `target_vault_name = aws_backup_vault.external_vault.name`?

Reply:
Works too! 

Created a separate example folder to highlight that use case scenario.